client/src/main/scala/com/all4journey/webapp/util/Fragments.scala

```diff
@@ -0,0 +1,151 @@
+      label(cls := "col-lg-3 control-label")("Street Address:"),
+      div(cls := "col-lg-8")(
+        div(id := "streetAddressHelpBlock"),
+        input(id := "streetAddress", name := "streetAddress", cls := "form-control", `type` := "text", value := address.getOrElse(emptyAddress).street.getOrElse(""))
+      )
+    ),
+    div(id := "stateDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("State:"),
+      div(cls := "col-lg-8")(
+        div(id := "stateHelpBlock"),
+        div(cls := "ui-select")(
+          select(id := "userState", name := "userState", cls := "form-control partOfStateList")(
+            option(value := address.getOrElse(emptyAddress).state.id, selected := "selected")(address.getOrElse(emptyAddress).state.description)
+          )
+        )
+      )
+    ),
+    div(id := "zipCodeDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("Zip code:"),
+      div(cls := "col-lg-8")(
+        div(id := "zipCodeHelpBlock"),
+        input(id := "zipCode", name := "zipCode", cls := "form-control partOfZipCodeList", `type` := "text", value := address.getOrElse(emptyAddress).zipCode)
+      )
+    )
+  )
+
+  def setUiViolationPrompts(violations: Set[Violation]) = {
+    if (!violations.isEmpty) {
+      //                      val content = dom.document.getElementById("content")
+      //                      for (v <- failureList)
+      //                        content.appendChild(p(v.description).render)
+      violations.foreach(violation =>
+        if (violation.description.getOrElse("").contains("street")) {
+          $("#streetAddressDiv").addClass("has-error")
+          val streetAddressHelpBlock = dom.document.getElementById("streetAddressHelpBlock")
+          streetAddressHelpBlock.appendChild(span(cls := "help-block")("invalid street address format").render)
+        }
+        else if (violation.description.getOrElse("").contains("state")) {
+          HtmlHelper.showHelpBlock("#stateDiv", "stateHelpBlock", "please choose a state")
+        }
+        else if (violation.description.getOrElse("").contains("zipCode")) {
+          HtmlHelper.showHelpBlock("#zipCodeDiv", "zipCodeHelpBlock", "invalid zip code format")
+
+        }
+      )
+    }
+  }
+
+  def validateAddress(address: Address): Set[Violation] = {
+    var violations = Set.empty[Violation]
+    val isFullAddressValid = validate(address)(Address.validatorWithStreet) match {
+      case ValidationSuccess   => true
+      case Failure(failureSet) =>
+        violations = failureSet
+        false
+    }
+
+    if (!isFullAddressValid) {
+      val partialAddressValidationResult = if (address.street.getOrElse("").isEmpty && !address.state.id.equals("NONE"))
+        validate(address)(Address.validatorNoStreetWithState)
+      else if (address.street.getOrElse("").isEmpty && address.state.id.equals("NONE") && (address.zipCode != null && !address.zipCode.isEmpty))
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/rjkernick/all4journey/file/2503283144/issues/source?bid=3246460&fileBranchId=3273764#l100)